### DISCUSSION TITLE
Recreating the Theme Generator

### DISCUSSION THREAD

**Discussion by Layendan** [saadeghi/daisyui#2050] [Q&A] [answered]
So I was trying to recreate the theme generator on the Daisyui website to let user create custom themes.
https://daisyui.com/theme-generator/

I tried a naive approach of setting the css variables as the value of the custom theme, but that didn't work.
<img width="456" alt="image" src="https://github.com/saadeghi/daisyui/assets/46062388/e870162f-944c-4c41-be97-2ba4c8953e88">
<img width="172" alt="image" src="https://github.com/saadeghi/daisyui/assets/46062388/86f47a08-831a-466e-aced-da82805e5e1f">
<img width="188" alt="image" src="https://github.com/saadeghi/daisyui/assets/46062388/84a925bb-2910-4abc-91fc-4629236550c2">

Does anyone have ideas on how to implement this, or how the website does it?

Thank you in advance for your help!

**Comment by saadeghi** [comment, marked as answer]
The CSS variable for colors are HSL values like `hsl(var(--p))`  

And this is what a theme looks like: https://unpkg.com/browse/daisyui@3.1.6/dist/themes.css

So if you have a color picker, you can set HSL values to CSS variable like `--p: 259 94% 51%`

Use this table as a reference https://daisyui.com/docs/colors/

Let me know if you have any questions

**Reply by Layendan** [reply]
Thank you so much. I completely forgot that the values were hsl
